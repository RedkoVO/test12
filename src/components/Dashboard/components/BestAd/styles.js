import gC from '../../../../constants'

export default theme => ({
  root: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
    marginTop: '40px',
    width: '100%',
    '-ms-align-items': 'center',

    [theme.breakpoints.down(gC.BRECKPOINT.MD)]: {
      '-ms-align-items': 'flex-start',
      alignItems: 'flex-start',
      justifyContent: 'center',
      flexWrap: 'wrap',
      marginTop: 0
    }
  },
  bestItem: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    '-ms-align-items': 'center',

    [theme.breakpoints.down(gC.BRECKPOINT.MD)]: {
      marginRight: '20px',
      marginBottom: '20px'
    }
  },
  titleWrap: {
    width: '100%',
    paddingLeft: '40px',
    boxSizing: 'border-box',

    [theme.breakpoints.down(gC.BRECKPOINT.MD)]: {
      display: 'flex',
      justifyContent: 'center',
      paddingLeft: 0
    }
  },
  bestTitle: {
    fontSize: '37px',
    fontWeight: 700,
    color: '#5e7cff',
    textAlign: 'center',
    marginBottom: '0px',
    textDecoration: 'none',
    margin: '0 auto'
  },
  bestContent: {
    display: 'flex',
    flexDirection: 'column',
    '-ms-align-items': 'center',
    alignItems: 'center',
  },
  bestImg: {
    display: 'block',
    borderRadius: '15px',
    width: '335px',
    height: '225px',
    boxShadow: '0px 10px 20px 0px rgba(0,0,0,0.2)',
    marginBottom: '25px'
  },
  bestItemImg: {
    width: '100%',
    height: '100%',
    borderRadius: '15px'
  },
  bestLink: {
    fontSize: '22px',
    fontWeight: 700,
    color: '#575757',
    textAlign: 'center',
    marginBottom: '0px',
    textDecoration: 'none',
    margin: '0 auto',

    '& span': {
      fontSize: '37px'
    }
  }
})