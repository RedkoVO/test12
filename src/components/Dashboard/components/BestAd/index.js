import React from 'react'
import PropTypes from 'prop-types'
import withStyles from '@material-ui/core/styles/withStyles'

import styles from './styles'

const BestAd = ({ classes, bestAds }) => (
  <div className={classes.root}>
    {bestAds.map(item => (
      <div className={classes.bestItem} key={item.id}>
        <div className={classes.titleWrap}>
          <a href="/" className={classes.bestTitle}>{item.title}</a>
        </div>
        <div className={classes.bestContent}>
          <a href="/" className={classes.bestImg}>
            <img src={item.img} alt="" className={classes.bestItemImg} />
          </a>
          <a href="/" className={classes.bestLink}>{item.titleLink}: <br /><span>{item.nameLink}</span></a>
        </div>
      </div>
    ))}
  </div>
)

BestAd.propTypes = {
  classNamees: PropTypes.object,
  bestAds: PropTypes.array
}

export default withStyles(styles)(BestAd)