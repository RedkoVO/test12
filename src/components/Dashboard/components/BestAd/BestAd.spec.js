import * as React from 'react'
import renderer from 'react-test-renderer'

import BestAd from './'

import BestGameLogo from '../../../../assets/images/best-game.png'

const data = [
  { id: 1, title: 'Games', img: BestGameLogo, titleLink: 'Game of the day', nameLink: 'Smash Champs' },
  { id: 2, title: 'Shop', img: BestGameLogo, titleLink: 'Product of the day', nameLink: 'Blue cool Jelly' }
]

it('renders correctly', () => {
  const tree = renderer.create(<BestAd bestAds={data} />).toJSON()
  expect(tree).toMatchSnapshot()
})