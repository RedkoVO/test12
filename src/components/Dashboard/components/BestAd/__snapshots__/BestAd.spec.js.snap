// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="BestAd-root-1"
>
  <div
    className="BestAd-bestItem-2"
  >
    <div
      className="BestAd-titleWrap-3"
    >
      <a
        className="BestAd-bestTitle-4"
        href="/"
      >
        Games
      </a>
    </div>
    <div
      className="BestAd-bestContent-5"
    >
      <a
        className="BestAd-bestImg-6"
        href="/"
      >
        <img
          alt=""
          className="BestAd-bestItemImg-7"
          src="best-game.png"
        />
      </a>
      <a
        className="BestAd-bestLink-8"
        href="/"
      >
        Game of the day
        : 
        <br />
        <span>
          Smash Champs
        </span>
      </a>
    </div>
  </div>
  <div
    className="BestAd-bestItem-2"
  >
    <div
      className="BestAd-titleWrap-3"
    >
      <a
        className="BestAd-bestTitle-4"
        href="/"
      >
        Shop
      </a>
    </div>
    <div
      className="BestAd-bestContent-5"
    >
      <a
        className="BestAd-bestImg-6"
        href="/"
      >
        <img
          alt=""
          className="BestAd-bestItemImg-7"
          src="best-game.png"
        />
      </a>
      <a
        className="BestAd-bestLink-8"
        href="/"
      >
        Product of the day
        : 
        <br />
        <span>
          Blue cool Jelly
        </span>
      </a>
    </div>
  </div>
</div>
`;
